### DISCUSSION TITLE
`skore` will have many dependencies as it will offer a wide range of features

### DISCUSSION THREAD

**Discussion by sylvaincom** [probabl-ai/skore#580] [Ideas]
#### Problem

In the long run, `skore` will be a buffet for data scientists and offer a wide variety of features. I am wondering whether putting all of those in one single python package is a good idea.

Indeed, scikit-learn rarely bugs because it has low dependencies and does not require many installs. In `skore`, you can do the dashboard but also just use `skore.cross_validate`. What if some users just want to use `skore.cross_validate`: they must beware of installing the frontend and backend (I see some warning about UTF-8 for Windows users) even if they do not need intend to use the dashboard.

This is a problem for the long run, but might be worth thinking about now.

### Describe the solution you'd like

On the long run:
1. Maybe split into several packages?
1. Maybe put the dashboard in extras of the same package?

**Comment by tuscland** [comment]
Hello Sylvain,

You are right, and this begs the question as to how to best manage dependencies and when/if to split the library in several parts. It is an engineering choice that should be based on more user feedback. Let's close this for now because there is nothing actionnable right now.

cc @thomass-dev

**Comment by sylvaincom** [comment]
Converted to discussion
